src/library/plugin/pluginType.ts

```diff
@@ -17,60 +18,38 @@ export type PluginType = {
 export interface IMinAuthPlugin<PublicInputsArgs, Output> {
   verifyAndGetOutput(
     publicInputArgs: PublicInputsArgs,
-    serializedProof: JsonProof): Promise<undefined | Output>;
+    serializedProof: JsonProof): Promise<Output>;
 
+  readonly publicInputArgsSchema: z.ZodType<PublicInputsArgs>;
 
-  checkOutputValidity(output: Output): Promise<boolean>;
+  // FIXME(Connor): I still have some questions regarding the validation functionality.
+  // In particular, what if a plugin want to invalidate the proof once the public inputs change?
+  // We have to at least pass PublicInputsArgs.
+  //
+  // checkOutputValidity(output: Output): Promise<boolean>;
 
-  readonly customRoutes: Map<string, RequestHandler>;
+  readonly customRoutes: Record<string, RequestHandler>;
 
   readonly verificationKey: string;
 }
```

Comment:
I think we want the plugin to return the prover part if asked. Not sure about it, what do you think?

Reply:
Hmm, I don't think this is what the server-side plugins should be concerned about.